pkg/agent/ipassigner/responder/arp_responder.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package responder
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/mdlayher/arp"
+	"github.com/mdlayher/ethernet"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+type arpResponder struct {
+	iface       *net.Interface
+	conn        *arp.Client
+	assignedIPs sets.String
+	mutex       sync.Mutex
```

Comment:
sync.Map is indeed a simpler choice here, but I know you need a mutex in ndpResponder for other reasons, so maybe you prefer to use the same way for both.

Reply:
Yes. I think the mutex here is better for consistency.